matching-notification-api/src/main/java/org/phenotips/matchingnotification/storage/internal/DefaultMatchStorageManager.java

```diff
@@ -138,125 +103,166 @@
     private Logger logger = LoggerFactory.getLogger(DefaultMatchStorageManager.class);
 
     @Override
-    public List<PatientMatch> getMatchesToBePlacedIntoNotificationTable(List<PatientSimilarityView> inputMatches)
+    public Map<PatientSimilarityView, PatientMatch>
+        saveLocalMatches(Collection<? extends PatientSimilarityView> similarityViews, String patientId)
     {
-        List<PatientMatch> matches = new LinkedList<>();
-        for (PatientSimilarityView similarityView : inputMatches) {
-            PatientMatch match = new DefaultPatientMatch(similarityView, null, null);
-
-            // filter out matches owned by the same user(s), as those are not shown in matching notification anyway
-            // and they break match count calculation if they are included
-            if (match.getReference().getEmails().size() > 0 && CollectionUtils.isEqualCollection(
-                match.getReference().getEmails(), match.getMatched().getEmails())) {
-                continue;
-            }
+        Predicate<PatientMatch> filterOutSameOwnerMatches = match -> {
```

Comment:
why do we need this again after [this](https://github.com/phenotips/patient-network/blob/master/similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java#L216-L220)?

Reply:
Ah, this is a regression that we have it "there". The filter is only supposed to be here. I will fix that